src/GenericContainer/include/BipedalLocomotionControllers/GenericContainer/Vector.h

```diff
@@ -0,0 +1,354 @@
+
+    Vector(iDynTree::Span<T> span, resize_function_type resizeLambda)
+    {
+        m_span = span;
+        m_resizeLambda = resizeLambda;
+    }
+
+    Vector(iDynTree::Span<T> span)
+    {
+        m_span = span;
+        m_resizeLambda = [span](index_type size){unused(size); return span;};
+    }
+
+    ~Vector() = default;
+
+    Vector(const Vector<T>& other) = delete; //It would not be clear if you are copying the pointers or the content.
+
+    Vector(Vector<T>&& other)
+    {
+        m_span = other.m_span;
+        m_resizeLambda = other.m_resizeLambda;
+    }
+
+    bool clone(const Vector<T>& other)
+    {
+        if (size() != other.size())
+        {
+            if (!resizeVector(other.size()))
+            {
+                std::cerr << "[GenericContainer::Vector] Failed to resize. Copy aborted" << std::endl;
+                return false;
+            }
+        }
+
+        for (index_type i = 0; i < size(); ++i)
+        {
+            this->operator[](i) = other[i];
+        }
+
+        return true;
+    }
+
+    void operator=(const Vector<T>& other)
+    {
+        clone(other);
+    }
+
+    void operator=(Vector<T>&& other)
+    {
+        clone(other);
+    }
+
+
+    bool resizeVector(index_type newSize)
+    {
+        m_span = m_resizeLambda(newSize);
+        return m_span.size() == newSize;
+    }
+
+    void resize(index_type newSize)
+    {
+        bool ok = resizeContainer(newSize);
+        assert(ok);
+        unused(ok);
+    }
+
+    index_type size() const
+    {
+        return m_span.size();
+    }
+
+    bool empty() const {
+        return m_span.empty();
+    }
+
+    const_reference operator[](index_type idx) const
+    {
+        return m_span[idx];
+    }
+
+    reference operator[](index_type idx)
+    {
+        return m_span[idx];
+    }
+
+    element_type getVal(index_type idx) const
+    {
+        return this->operator[](idx);
+    }
+
+    bool setVal(index_type idx, element_type val)
+    {
+        if (idx >= 0 && idx < size())
+        {
+            this->operator[](idx) = val;
+            return true;
+        }
+        return false;
+    }
+
+    const_reference at(index_type idx) const
+    {
+        return this->operator[](idx);
+    }
+
+    reference at(index_type idx)
+    {
+        return this->operator[](idx);
+    }
+
+    const_reference operator()(index_type idx) const
+    {
+        return this->operator[](idx);
+    }
+
+    reference operator()(index_type idx)
+    {
+        return this->operator[](idx);
+    }
+
+    pointer data() const
+    {
+        return m_span.data();
+    }
+
+    // [span.iter], span iterator support
+    iterator begin()
+    {
+        return m_span.begin();
+    }
+    iterator end()
+    {
+        return m_span.end();
+    }
+
+    const_iterator cbegin() const
+    {
+        return m_span.cbegin();
+    }
+
+    const_iterator cend() const
+    {
+        return m_span.cend();
+    }
```

Comment:
I would add also the `const` version of `begin` and `end`
i.e. 
```cpp
    const_iterator begin() const
    {
        return m_span.cbegin();
    }
    const_iterator end() const
    {
        return m_span.cend();
    }
```

This will allow using Range-based for loop for the container.

Similar issue: https://github.com/robotology/yarp/issues/2179 and associated PR https://github.com/robotology/yarp/pull/2182

Reply:
Added in https://github.com/dic-iit/bipedal-locomotion-controllers/pull/29/commits/1a117600b0ae7245a88ab716c719101fa2161db5